### PR TITLE
Build-only Associations

### PR DESCRIPTION
**Pull request by kspaans**
Thanks to #19 for inspiration for this PR. I'm happy to rebase this onto that branch if that's more appropriate.

This refactors the code a little and adds build-only associations to the factory -- useful for unit testing.

### CODE REVIEW

**Comment by kspaans** [comment]
Whoops, missed #50! That code also works for my unit tests. @aexmachina let me know if any of these changes are useful and I'll try to merge them in.

File: index.js [hunk 1]
```diff
@@ -47,37 +47,76 @@
     factory.create = builderProxy('create');
     factory.createMany = builderProxy('createMany');
 
-    factory.assoc = function(name, key, attrs) {
+    factory.assocHelper = function(constructor, name, key, attrs) {
```
Comment (aexmachina) [inline index.js]: This is good :) Maybe fn is better than constructor?

File: index.js [hunk 2]
```diff
@@ -47,37 +47,76 @@
...
+    factory.assocMany = function (name, key, num, attrsArray) {
+      var args = parseAssocManyArgs.apply(null, arguments);
+      return factory.assocManyHelper(factory.createMany, args);
+    };
+
+    factory.assocBuildMany = function(name, key, num, attrsArray) {
+      var args = parseAssocManyArgs.apply(null, arguments);
+      return factory.assocManyHelper(factory.buildMany, args);
+    };
+
```
Comment (aexmachina) [inline index.js]: Yes this also is good.

**Comment by aexmachina** [comment]
There's been some movement in master since you started. I hope you don't have trouble merging because I like these changes. Nice.

**Comment by kspaans** [comment]
Thanks @aexmachina, it should be doable. I'll take a look at merging things later today.

**Comment by aexmachina** [comment]
I'd like to get these changes into v4.0-beta, which I released today. Thanks!

**Comment by aexmachina** [comment]
Actually, we're working towards 3.2.0-beta because all these changes should be backwards-compatible. Are you able to get this PR ready for merge?

**Comment by kspaans** [comment]
Sure, I'll re-spin against the `3.2.0-beta` tag. Sorry for the delay, I didn't realize it was going to be a long weekend!

**Comment by aexmachina** [comment]
Closing this in favour of #56. We'll be doing a v4-beta release shortly, it'd be great for you to download and test!

**Pull request closed without merging**